src/app/plugins/commandsearch.plugin.ts

```diff
@@ -0,0 +1,61 @@
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IMessage, IPlugin } from '../../common/types';
+
+export class CommandSearchPlugin extends Plugin {
+  public name: string = 'Command Search';
+  public description: string = 'Search our commands on a key word or phrase';
+  public usage: string = 'commands <word or phrase>';
+  public pluginAlias = ['searchcommands', 'grep'];
+  public permission: ChannelType = ChannelType.Bot;
+  public commandPattern: RegExp = /[^]+/;
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public async execute(message: IMessage, args: string[]) {
+    const query = args.join(' ');
+
+    // For every plugin, evaluate it's match
+    const results = Object.entries(this.container.pluginService.plugins).reduce(
+      (results: string[], [pluginName, plugin]) => {
+        if (
+          plugin.permission === ChannelType.Admin ||
+          plugin.permission === ChannelType.Staff ||
+          !this._grep(plugin, query)
+        ) {
```

Comment:
this doesnt quite make it so that the response always only returns commands that are available in the calling channel

Reply:
That was a different PR, this command is only in #bot_commands

Reply:
Yeah, we want users to be able to query all commands.